app/src/main/java/eu/kanade/tachiyomi/source/online/english/Mangadex.kt

```diff
@@ -324,6 +326,101 @@ open class Mangadex(override val lang: String, private val internalLang: String,
+        val document = response.asJsoup()
+
+        val follows = document.select(followSelector()).map { element ->
+            followFromElement(element)
+        }
+
+        val hasNextPage = followsNextPageSelector()?.let { selector ->
+            document.select(selector).first()
+        } != null
+
+        return Pair(follows, hasNextPage)
+    }
+
+    private fun getFollowsClient(filters: FilterList): OkHttpClient { // TODO reuse/rename `getSearchClient()`?
+        filters.forEach { filter ->
+            when (filter) {
+                is R18 -> {
+                    return when (filter.state) {
+                        1 -> clientBuilder(ALL)
+                        2 -> clientBuilder(ONLY_R18)
+                        3 -> clientBuilder(NO_R18)
+                        else -> clientBuilder()
+                    }
+                }
+            }
+        }
+        return clientBuilder()
+    }
+
+    fun followsListRequest(page: Int, filters: FilterList): Request {
+
+        // Format `/follows/manga/$status[/$sort[/$page]]`
+        val url = HttpUrl.parse("$baseUrl/follows/manga/0")!!.newBuilder() // Gets regardless of follow status. TODO add filter?
+                .addQueryParameter("p", page.toString())
+
+        filters.forEach { filter ->
+            when (filter) {
+                is TextField -> url.addQueryParameter(filter.key, filter.state)
+                is SortFilter -> {
+                    if (filter.state != null) {
+                        if (filter.state!!.ascending) {
+                            url.addQueryParameter("s", sortables[filter.state!!.index].second.toString())
+                        } else {
+                            url.addQueryParameter("s", sortables[filter.state!!.index].third.toString())
+                        }
+                    }
+                }
+            }
+        }
+
+
+        return GET(url.toString(), headersBuilder().build())
+    }
+
+    private fun followSelector() = "div.manga-entry"
+
+    private fun followFromElement(element: Element): Pair<SManga, Int> { // TODO can follow status be added to `SManga`?
+        val manga = SManga.create()
+
+        element.select("a.manga_title").first().let {
+            val url = modifyMangaUrl(it.attr("href"))
+            manga.setUrlWithoutDomain(url)
+            manga.title = it.text().trim()
+        }
+
+        manga.thumbnail_url = formThumbUrl(manga.url)
+
+        val followStatus = element
+                .select("button.btn.btn-success.dropdown-toggle:has(span.fas.fa-fw:not(.fa-star))")
+                .first() //Select the first dropdown that doesn't contain a rating element. Note: `:has()` is not currently supported in browsers
+                .text()
+                .trim()
+                .let { FOLLOW_STATUS_LIST.getValue(it) } // TODO: There's probably a better way to convert the value to an `Int` (If even needed)
```

Comment:
Making FOLLOW_STATUS_LIST  a listOf like below.  Allows you to do like below (might not be 100% )

FOLLOW_STATUS_LIST.first(pair -> pair.first == it).second

but basically take the first pair out of the list that matches the follow, then take the integer from that pair